packages/opentelemetry/src/init.ts

```diff
@@ -218,12 +219,14 @@ function getMetricExporter(config: OpenTelemetryConfig): PushMetricExporter | nu
 }
 
 function getSpanProcessor(config: OpenTelemetryConfig): SpanProcessor | null {
+  if (!config.openTelemetryEnabled) return new NoopSpanProcessor();
+
   if (typeof config.openTelemetrySpanProcessor === 'object') {
     return config.openTelemetrySpanProcessor;
   }
 
   const traceExporter = getTraceExporter(config);
-  if (!traceExporter) return null;
+  if (!traceExporter) return new NoopSpanProcessor();
```

Comment:
This ends up being the default in OTel internals. I opted to just do this here to make things really explicit.